src/object_store/src/object/s3.rs

```diff
@@ -591,8 +627,17 @@ impl S3ObjectStore {
         }
 
         // Retry 3 times if we get server-side errors or throttling errors
-        let sdk_config_loader =
+        let mut sdk_config_loader =
             aws_config::from_env().retry_config(RetryConfig::standard().with_max_attempts(4));
+        #[cfg(not(madsim))]
+        {
+            // Customize connector
```

Comment:
```rust
    let mut http = hyper::client::HttpConnector::new();
        http.set_keepalive(Some(Duration::from_secs(600)));
        http.enforce_http(false);
        let smithy_connector = aws_smithy_client::hyper_ext::Adapter::builder()
            .hyper_builder(hyper::client::Builder::default())
            .connector_settings(ConnectorSettings::builder().build())
            .build(http);
        // Retry 3 times if we get server-side errors or throttling errors
        let sdk_config_loader = aws_config::from_env()
            .retry_config(RetryConfig::standard().with_max_attempts(4))
            .http_connector(smithy_connector);
```
We have not changed ConnectorSettings, can we use the above code to simplify customized_connector?